src/include/miopen/avgpool/problem_description.hpp

```diff
@@ -0,0 +1,186 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <miopen/problem_description_base.hpp>
+#include <miopen/activ.hpp>
+#include <miopen/tensor.hpp>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace avgpool {
+
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(const bool count_include_pad_, const int32_t divisor_override_)
+        : count_include_pad(count_include_pad_), divisor_override(divisor_override_)
+    {
+        if(divisor_override < 0)
+        {
+            MIOPEN_THROW(miopenStatusBadParm, "AvgPool: divisor_override must be non-negative.");
+        }
+    }
+
+protected:
+    bool count_include_pad;
+    int32_t divisor_override;
+};
+
+struct FwdProblemDescription : ProblemDescription
+{
+    FwdProblemDescription(const TensorDescriptor& inputDesc_,
+                          const TensorDescriptor& outputDesc_,
+                          const bool count_include_pad_,
+                          const int32_t divisor_override_)
+        : ProblemDescription(count_include_pad_, divisor_override_),
+          inputDesc(inputDesc_),
+          outputDesc(outputDesc_)
+    {
+        IsValidLength();
+        IsSameType();
+        IsValidDims();
+    }
+
+    auto GetInputDesc() const { return inputDesc; }
+    auto GetOutputDesc() const { return outputDesc; }
```

Comment:
Recommend this to avoid copy `inputDesc` and `outputDesc` when calling these 2 functions. same with backward
```suggestion
    const TensorDescriptor& GetInputDesc() const { return inputDesc; }
    const TensorDescriptor& GetOutputDesc() const { return outputDesc; }
```

Reply:
resolved